src/scraper.rs

```diff
@@ -0,0 +1,390 @@
+            .children()
+            .filter(|node| node.is(Name("p")))
+            .map(Paragraph)
+            .collect();
+        if paragraphs.is_empty() {
+            bail!("summary contains no paragraphs");
+        }
+
+        let articles = story
+            .find(Class("feature-thumbs-wrapper"))
+            .next()
+            .ok_or_else(|| {
+                anyhow!("cannot query linked stories container (class=feature-thumbs-wrapper)")
+            })?;
+
+        let articles = articles.find(Class("feature-thumbs"));
+        let articles: Vec<_> = articles.map(|node| {
+            let title = node.find(Class("news-title")).next()
+                .and_then(|node| node.find(Name("a")).next())
+                .map(|node| node.text())
+                .ok_or_else(|| anyhow!("cannot query linked article title (class=news-title > a)"))?
+                .trim().into();
+
+            let url = node.find(Class("read-more-story")).next()
+                .and_then(|node| node.find(Name("a")).next())
+                .and_then(|node| node.attr("href"))
+                .ok_or_else(|| anyhow!("cannot query linked article origin url (class=read-more-story > a.href)"))?
+                .to_owned();
+
+            let source = node.find(Class("news-source")).next()
+                .ok_or_else(|| anyhow!("cannot query news article source (class=news-source)"))?
+                .text();
+
+            let bias = node.find(Class("bias-image")).next()
+                .and_then(|node| node.first_child())
+                .filter(|node| node.is(Name("img")))
+                .and_then(|node| node.attr("title"))
+                .ok_or_else(|| anyhow!("cannot query news source political bias (class=bias-image > img.title)"))?;
+
+            if !bias.contains(':') || bias.is_empty() {
+                bail!("unexpected bias format: expected '<any>: <affiliation>'");
+            }
+
+            let side_shorthand = bias.split(':').last()
+                .ok_or_else(|| anyhow!("bug: unexpected bias format (pre-check passed, parsing failed)"))?
+                .trim();
+
+            let side = Side::from_str(side_shorthand)?;
+
+            let description = node.find(Class("news-body")).next()
+                .ok_or_else(|| anyhow!("cannot query article summary (class=news-body)"))?;
+            let paragraphs: Vec<_> = description.children()
+                .filter(|node| node.is(Name("p")))
+                .map(Paragraph)
+                .collect();
+
+            Ok(Article {
+                side,
+                source,
+                title,
+                summary: paragraphs,
+                url,
+            })
+        }).collect::<Result<_, _>>()?;
+
+        Ok(Story {
+            title,
+            summary: paragraphs,
+            articles,
+            datetime,
+        })
+    }
+}
+
+fn normalize_allsides_url(relative_url: &str) -> String {
+    format!("https://www.allsides.com{}", relative_url)
```

Comment:
```suggestion
    ["https://www.allsides.com{}", relative_url].concat()
```
Also this implies that relative URL begins with `/`. Maybe put that slash here?
Also I saw ALLSIDES_MAINPAGE_URL or something like this. May we get rid of duplication?